### ISSUE TITLE
Allow multiple policies per Thing

### ISSUE THREAD

**Issue by vhdirk** [eclipse/ditto#991]
As discussed with @thjaeckle over gitter, we're looking for a way to assign multiple policies to a Thing. That would make granular permission control a lot easier to implement.

Below a simple demonstration of what we'd like to achieve:
![multiple-policies](https://user-images.githubusercontent.com/1424486/109845953-f59d8800-7c4d-11eb-9ee1-c632de1f3c83.png)
Take for example a situation where Things are installed at 2 physical locations. Operators on site would have permission to view properties of the Thing, but not modify (Teams A and B). Overseeing all operations is Team X, which has permission to modify/delete every Thing. Additionally, there's a maintenance team (Team C) that can travel to both physical locations, but only has permission to change specific features of Things (not necessarily restricted that much), for a specific subset of Things.

To achieve something like that, it would be (for a ditto developer) a lot easier if things could be assigned multiple policies.

Compare that to what we currently would have to configure:
![single-policy](https://user-images.githubusercontent.com/1424486/109845958-f6361e80-7c4d-11eb-9778-ccded1331ced.png)

We understood that this may have a negative performance impact on the thingssearch index, as it would have to be rebuilt if a policy is modified. However, whether using multiple policies or not, that bottleneck is always going to be there.

**Comment by thjaeckle** [comment]
Just to clarify: #298 would also let you do exactly what you described, won't it?
Because I think that this would API-wise be a better enhancement than e.g. adding an optional `"additionalPolicyIds"` to the Thing JSON.

**Comment by vhdirk** [comment]
Yes, indeed. #298 would make this situation possible, but it has some drawbacks, too.

What we're proposing here would mean breaking the current API: a Thing would really just take an array of `PolicyId`s, rather than a single one. This would keep the possibility to filter things based on policyId. With the 'import' proposal made in #298, that would require an indirection: You'd first have to find all policies that import the one you're querying, and then match those with the actual devices.

**Comment by thjaeckle** [comment]
I'm afraid that breaking the JSON API is not a short- to midterm option, at least not until Ditto requires other JSON breakages in the "thing JSON" which would justify an "API schema version 3" (e.g. `/api/3` at HTTP level). 
With Ditto 2.0 we remove "API schema version 1" (ACL based) and maintaining several schema versions at the same time is a huge amount of additional maintenance work.

Additionally, as policies may also be used by other entities than things (e.g. also policies itself), all those other entities would also need to support containing a list of policyIds. Better in terms of re-use from my point of view is to make the policy more mighty (e.g. by the imports mentioned in #298).

Filtering things based on policyId is for me another use case which should be tracked separately.
Technically this would probably be another search index.

**Comment by thjaeckle** [comment]
Closing as duplicate of #298
